components/modal/modal-action.tsx

```diff
@@ -1,84 +1,60 @@
-import React, { MouseEvent, useMemo } from 'react'
+import React, { MouseEvent } from 'react'
 import withDefaults from '../utils/with-defaults'
-import useTheme from '../styles/use-theme'
+import Button from '../button'
+import { ButtonColors, ButtonVariants, NormalSizes } from '../utils/prop-types'
 import { useModalContext } from './modal-context'
 
 type ModalActionEvent = MouseEvent<HTMLButtonElement> & {
   close: () => void
 }
 
 interface Props {
-  className?: string
-  passive?: boolean
+  variant?: ButtonVariants
+  color?: ButtonColors
+  size?: NormalSizes
+  ghost?: boolean
+  loading?: boolean
+  shadow?: boolean
+  auto?: boolean
+  effect?: boolean
   disabled?: boolean
+  htmlType?: React.ButtonHTMLAttributes<any>['type']
+  icon?: React.ReactNode
+  iconRight?: React.ReactNode
   onClick?: (event: ModalActionEvent) => void
+  className?: string
 }
 
 const defaultProps = {
-  className: '',
-  passive: false,
+  variant: 'line' as ButtonVariants,
+  color: 'primary' as ButtonColors,
+  size: 'medium' as NormalSizes,
+  htmlType: 'button' as React.ButtonHTMLAttributes<any>['type'],
+  ghost: false,
+  loading: false,
+  shadow: false,
+  auto: false,
+  effect: true,
   disabled: false,
+  className: '',
 }
 
 type NativeAttrs = Omit<React.ButtonHTMLAttributes<any>, keyof Props>
 export type ModalActionProps = Props & typeof defaultProps & NativeAttrs
```

Comment:
export type without type of defaultProps

Reply:
done